### PR TITLE
Method take for datastream

### PR DESCRIPTION
**Pull request by samedii**
Method `Datastream.take`. Almost an alias for `Datastream.sample_proportion`.

### CODE REVIEW

File: datastream/datastream.py [hunk 1]
```diff
@@ -213,6 +216,16 @@ def sample_proportion(
...
 
+    def take(
+        self: Datastream[T],
+        n_samples: int,
+    ) ->  Datastream[T]:
+        '''
+        Like :func:`Datastream.sample_proportion` but specify the number of
+        samples instead of a proportion.
+        '''
+        return self.sample_proportion(min(1, n_samples / len(self)))
```
Comment (Jim-Holmstroem) [inline datastream/datastream.py:227]: We allow empty datastreams right? (even if its not super useful)
I know its a degenerate case, but this crashes kinda gnarly if `len(self) == 0`
Also I think `len(self) == 0` and `n_samples == 0` should return an empty datastream

File: datastream/samplers.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def __init__(self, length, proportion=1.0, replacement=False):
             replacement=replacement,
             sampler=torch.utils.data.WeightedRandomSampler(
                 torch.ones(length).double(),
-                num_samples=int(length * proportion),
+                num_samples=max(int(length * proportion), 1),
```
Comment (Jim-Holmstroem) [inline datastream/samplers.py:27]: Ties back to the earlier comment about zero lengths etc
Not sure how everything reacts, but usually things are more stable when even degenerate cases makes sense and works

**Review by Jim-Holmstroem — CHANGES_REQUESTED** [review]

File: datastream/datastream.py (hunk 1 above)
Comment (samedii) [reply datastream/datastream.py:227]: We do not support empty datastreams. The most common case where this fails for me has been when merging empty datastreams

File: datastream/samplers.py (hunk 2 above)
Comment (samedii) [reply datastream/samplers.py:27]: We can try to handle empty datastreams better. Not sure how easy it would be

Comment (Jim-Holmstroem) [reply datastream/samplers.py:27]: If its not supported then it should crash early, so crash on init with `ValueError` for example

**Review by Jim-Holmstroem — APPROVED** [review]

File: datastream/datastream.py (hunk 1 above)
Comment (samedii) [reply datastream/datastream.py]: Added check and tests

File: datastream/samplers.py (hunk 2 above)
Comment (samedii) [reply datastream/samplers.py:27]: Added check

**Review by Jim-Holmstroem — APPROVED** [review]

**Review by FelixAbrahamsson — APPROVED** [review]

**Pull request merged**